File: python/program/Pegawai.py
from Orang import Orang

# subclass Pegawai
class Pegawai(Orang):

    def __init__(self, noHp: str, email: str, alamat: str,
                 nama: str, usia: int, tanggalLahir: str,
                 kodePegawai: str, jabatan: str, gaji: str, shift: str):
        # panggil constructor parent (Orang)
        super().__init__(noHp, email, alamat, nama, usia, tanggalLahir)
        # atribut khusus Pegawai
        self._kodePegawai = str(kodePegawai)
        self._jabatan = str(jabatan)
        self._gaji = str(gaji)       # gaji disimpan string biar konsisten dengan saldo
        self._shift = str(shift)

    # Getter & Setter Kode Pegawai
    def getKodePegawai(self) -> str:
        return self._kodePegawai

    def setKodePegawai(self, kodePegawai: str) -> None:
        self._kodePegawai = str(kodePegawai)

    # Getter & Setter Jabatan
    def getJabatan(self) -> str:
        return self._jabatan

    def setJabatan(self, jabatan: str) -> None:
        self._jabatan = str(jabatan)

    # Getter & Setter Gaji
    def getGaji(self) -> str:
        return self._gaji

    def setGaji(self, gaji: str) -> None:
        self._gaji = str(gaji)

    # Getter & Setter Shift
    def getShift(self) -> str:
        return self._shift

    def setShift(self, shift: str) -> None:
        self._shift = str(shift)

File: java/program/Pegawai.java
public class Pegawai extends Orang {
    /* Class Pegawai ini turunan dari class Orang,
    Di sini saya tambahin atribut khusus untuk pegawai: kode pegawai, jabatan, gaji, dan shift. */
    private String kodePegawai;
    private String jabatan;
    private String gaji;
    private String shift;

    // Constructor ini dipakai buat ngisi data pegawai.
    // Pertama, constructor parent (Orang) dipanggil dengan super(), lalu atribut khusus pegawai diisi.
    public Pegawai(String noHp, String email, String alamat,
                   String nama, int usia, String tanggalLahir,
                   String kodePegawai, String jabatan, String gaji, String shift) {
        super(noHp, email, alamat, nama, usia, tanggalLahir);
        this.kodePegawai = kodePegawai;
        this.jabatan = jabatan;
        this.gaji = gaji;
        this.shift = shift;
    }

    // Getter dan setter berikut dipakai untuk akses atau ubah
    // data khusus pegawai tadi: kode pegawai, jabatan, gaji, dan shift.
    public String getKodePegawai() {
        return kodePegawai;
    }

    public void setKodePegawai(String kodePegawai) {
        this.kodePegawai = kodePegawai;
    }

    public String getJabatan() {
        return jabatan;
    }

    public void setJabatan(String jabatan) {
        this.jabatan = jabatan;
    }

    public String getGaji() {
        return gaji;
    }

    public void setGaji(String gaji) {
        this.gaji = gaji;
    }

    public String getShift() {
        return shift;
    }

    public void setShift(String shift) {
        this.shift = shift;
    }
}
